Makefile

```diff
@@ -102,6 +108,12 @@ $(BINDIR)/deepcopy-gen: .init cmd/libs/go2idl/deepcopy-gen
 	  -i $(SC_PKG)/pkg/apis/servicecatalog,$(SC_PKG)/pkg/apis/servicecatalog/v1alpha1 \
 	  --bounding-dirs github.com/kubernetes-incubator/service-catalog \
 	  -O zz_generated.deepcopy
+	$(DOCKER_CMD) $(BINDIR)/conversion-gen --v 1 --logtostderr \
+	  -i $(SC_PKG)/pkg/apis/servicecatalog,$(SC_PKG)/pkg/apis/servicecatalog/v1alpha1 \
+	  --extra-peer-dirs k8s.io/kubernetes/pkg/api,k8s.io/kubernetes/pkg/api/v1,k8s.io/kubernetes/pkg/apis/meta/v1,k8s.io/kubernetes/pkg/conversion,k8s.io/kubernetes/pkg/runtime \
+	  -O zz_generated.conversion
+	  # the previous three directories will be changed from kubernetes to apimachinery in the future
+	$(DOCKER_CMD) $(HACKDIR)/update-codecgen.sh
```

Comment:
It generates a YAML parser for all the versioned API packages. @jpeeler would you add a comment to the top of the codecgen script clarifying what it does?

Also, FYI, at some point we will drop the need to generate these YAML parsers.  Unfortunately you have to have a custom codec for all your types if you have it for any types, which the k8s types do.  We'll drop them sometime in the future because they were originally introduced to get around a hotspot with using a generic YAML parser before we did protobuf for traffic between system components and the API server.